ceph/ceph_admin/helper.py

```diff
@@ -545,20 +545,23 @@ def generate_rgw_spec(self, spec):
         if node_names:
             spec["placement"]["hosts"] = self.get_hostnames(node_names)
 
-        if spec["spec"].get("rgw_frontend_ssl_certificate") == "create-cert":
-            subject = {
-                "common_name": spec["placement"]["hosts"][0],
-                "ip_address": self.cluster.get_node_by_hostname(
-                    spec["placement"]["hosts"][0]
-                ).ip_address,
-            }
-            key, cert, ca = generate_self_signed_certificate(subject=subject)
-            pem = key + cert + ca
-            cert_value = "|\n" + pem
-            spec["spec"]["rgw_frontend_ssl_certificate"] = "\n    ".join(
-                cert_value.split("\n")
-            )
-            LOG.debug(pem)
+        if spec.get("spec", False):
+            if spec["spec"].get("rgw_frontend_ssl_certificate", False):
```

Comment:
@anrao19 
also the line previously here was
if spec["spec"].get("rgw_frontend_ssl_certificate") == "create-cert":

but you are modifying it to 
if spec["spec"].get("rgw_frontend_ssl_certificate", False):

is this intentionally done or by mistake?

Reply:
as of now in script we have rgw_frontend_ssl_certificate assigned only with value create-cert, so this was intentional